File: airbyte-db/db-lib/src/test/java/io/airbyte/db/check/impl/CommonDatabaseCheckTest.java
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.db.check.impl;

import io.airbyte.db.factory.DSLContextFactory;
import io.airbyte.db.factory.DataSourceFactory;
import javax.sql.DataSource;
import org.jooq.DSLContext;
import org.jooq.SQLDialect;
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeEach;
import org.testcontainers.containers.PostgreSQLContainer;

/**
 * Common test setup for database availability check tests.
 */
class CommonDatabaseCheckTest {

  protected static final long TIMEOUT_MS = 500L;

  protected PostgreSQLContainer<?> container;

  protected DataSource dataSource;

  protected DSLContext dslContext;

  @BeforeEach
  void setup() {
    container = new PostgreSQLContainer<>("postgres:13-alpine");
    container.start();

    dataSource = DataSourceFactory.create(container.getUsername(), container.getPassword(), container.getDriverClassName(), container.getJdbcUrl());
    dslContext = DSLContextFactory.create(dataSource, SQLDialect.POSTGRES);
  }

  @SuppressWarnings("PMD.SignatureDeclareThrowsException")
  @AfterEach
  void cleanup() throws Exception {
    DataSourceFactory.close(dataSource);
    dslContext.close();
    container.stop();
  }

}

File: airbyte-cdk/java/airbyte-cdk/core/src/test/kotlin/io/airbyte/cdk/db/check/impl/CommonDatabaseCheckTest.kt
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.cdk.db.check.impl

import io.airbyte.cdk.db.factory.DSLContextFactory
import io.airbyte.cdk.db.factory.DataSourceFactory
import javax.sql.DataSource
import org.jooq.DSLContext
import org.jooq.SQLDialect
import org.junit.jupiter.api.AfterEach
import org.junit.jupiter.api.BeforeEach
import org.testcontainers.containers.PostgreSQLContainer

/** Common test setup for database availability check tests. */
internal class CommonDatabaseCheckTest {
    protected var container: PostgreSQLContainer<*>? = null

    protected var dataSource: DataSource? = null

    protected var dslContext: DSLContext? = null

    @BeforeEach
    fun setup() {
        container = PostgreSQLContainer<Nothing>("postgres:13-alpine")
        container!!.start()

        dataSource =
            DataSourceFactory.create(
                container!!.username,
                container!!.password,
                container!!.driverClassName,
                container!!.jdbcUrl
            )
        dslContext = DSLContextFactory.create(dataSource, SQLDialect.POSTGRES)
    }

    @AfterEach
    @Throws(Exception::class)
    fun cleanup() {
        DataSourceFactory.close(dataSource)
        container!!.stop()
    }

    companion object {
        protected const val TIMEOUT_MS: Long = 500L
    }
}
